overlord/configstate/configcore/hostname.go

```diff
@@ -38,24 +38,23 @@ func init() {
 	config.RegisterExternalConfig("core", "system.hostname", getHostnameFromSystemHelper)
 }
 
-// We are conservative here and follow hostname(7). The hostnamectl
-// binary is more liberal but let's err on the side of caution for
-// now.
-var validHostnameRegexp = regexp.MustCompile(`^[a-z0-9][a-z0-9-]{0,62}(\.[a-z0-9-]{1,63})*$`).MatchString
+// The hostname can also be set via hostnamectl so we cannot be more strict
+// than hostnamectl itself.
+// See: systemd/src/basic/hostname-util.c:ostname_is_valid
+var validHostnameRegexp = regexp.MustCompile(`^[a-zA-Z0-9][a-zA-Z0-9-]{0,62}(\.[a-zA-Z0-9-]{1,63})*$`).MatchString
```

Comment:
so I can't find any situations where this won't allow something that hostnamectl allows, however there are a number of cases that hostnamectl doesn't allow that we will allow with this, for example these cases should be denied:
* `A.-.A`
* `A-.-A`
* `A-A-`
* `A.-`
* `A-.-`
* `A.-A`
* `A.-A`

but I think it's okay since being less strict than hostnamectl is okay for now I think